src/testhost.x86/DefaultEngineInvoker.cs

```diff
@@ -46,28 +48,157 @@ internal class DefaultEngineInvoker :
 
         private const string TelemetryOptedIn = "--telemetryoptedin";
 
+        private ITestRequestHandler requestHandler;
+
+        private IDataCollectionTestCaseEventSender dataCollectionTestCaseEventSender;
+
+        private IProcessHelper processHelper;
+
+
+        public DefaultEngineInvoker() : this(new TestRequestHandler(), DataCollectionTestCaseEventSender.Create(), new ProcessHelper())
+        {
+        }
+
+        internal DefaultEngineInvoker(ITestRequestHandler requestHandler,
+            IDataCollectionTestCaseEventSender dataCollectionTestCaseEventSender, IProcessHelper processHelper)
+        {
+            this.processHelper = processHelper;
+            this.requestHandler = requestHandler;
+            this.dataCollectionTestCaseEventSender = dataCollectionTestCaseEventSender;
+        }
+
         public void Invoke(IDictionary<string, string> argsDictionary)
         {
-            // Setup logging if enabled
-            if (argsDictionary.TryGetValue(LogFileArgument, out string logFile))
+            DefaultEngineInvoker.InitializeEqtTrace(argsDictionary);
+
+            if (EqtTrace.IsInfoEnabled)
             {
-                EqtTrace.InitializeVerboseTrace(logFile);
+                EqtTrace.Info("DefaultEngineInvoker.Invoke: Testhost process started with args :{0}",
+                    string.Join(",", argsDictionary));
+#if NET451
+                var appConfigText =
+ System.IO.File.ReadAllText(AppDomain.CurrentDomain.SetupInformation.ConfigurationFile);
+                EqtTrace.Info("DefaultEngineInvoker: Using Application Configuration: '{0}'", appConfigText);
+#endif
             }
-            else
+
+            this.SetParentProcessExitCallback(argsDictionary);
+
+            this.requestHandler.ConnectionInfo =
+                DefaultEngineInvoker.GetConnectionInfo(out var endpoint, argsDictionary);
```

Comment:
>endpoint [](start = 63, length = 8)

remove endpoint, move logging in GetConnectionInfo()

Reply:
Done.